device/cuda/src/finding/finding_algorithm.cu

```diff
@@ -498,21 +507,21 @@ finding_algorithm<stepper_t, navigator_t>::operator()(
                                   m_cfg.max_track_candidates_per_track),
          m_mr.main, m_mr.host, vecmem::data::buffer_type::resizable}};
 
-    m_copy->setup(track_candidates_buffer.headers);
-    m_copy->setup(track_candidates_buffer.items);
+    m_copy.setup(track_candidates_buffer.headers);
+    m_copy.setup(track_candidates_buffer.items);
 
     // @Note: nBlocks can be zero in case there is no tip. This happens when
     // chi2_max config is set tightly and no tips are found
     if (n_tips_total > 0) {
         nThreads = WARP_SIZE * 2;
         nBlocks = (n_tips_total + nThreads - 1) / nThreads;
-        kernels::build_tracks<<<nBlocks, nThreads>>>(
+        kernels::build_tracks<<<nBlocks, nThreads, 0, stream>>>(
             measurements, seeds_buffer, links_buffer, param_to_link_buffer,
             tips_buffer, track_candidates_buffer);
 
         CUDA_ERROR_CHECK(cudaGetLastError());
-        CUDA_ERROR_CHECK(cudaDeviceSynchronize());
     }
+    m_stream.synchronize();
```

Comment:
This should stay here for this PR.

But... at one point we'll need to experiment with removing these "final sync points" from the algorithms. Since in principle the returned buffer **can** wait for an asynchronous copy "on the caller's side" as well. (The buffer is moved back to the user. So the asynchronous copy will "find it".)

Reply:
OK